### PR TITLE
INTMDB-328: Update resource & data source to address change in go SDK that impacts resource_mongodbatlas_x509_authentication_database_user

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Update resource & data source to address change in go SDK that impacts resource_mongodbatlas_x509_authentication_database_user

Link to any related issue(s):
https://jira.mongodb.org/browse/INTMDB-328

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM!  Changed the APIx team review to see if it would select a new reviewer as Andrea is out.

**Review by fmenezes — APPROVED** [review]
LGTM

File: mongodbatlas/data_source_mongodbatlas_x509_authentication_database_user.go [hunk 1]
```diff
@@ -66,7 +67,8 @@ func dataSourceMongoDBAtlasX509AuthDBUserRead(ctx context.Context, d *schema.Res
 	username := d.Get("username").(string)
 
 	if username != "" {
-		certificates, _, err := conn.X509AuthDBUsers.GetUserCertificates(ctx, projectID, username)
+		options := &matlas.ListOptions{}
+		certificates, _, err := conn.X509AuthDBUsers.GetUserCertificates(ctx, projectID, username, options)
```
Comment (gssbzn) [inline mongodbatlas/data_source_mongodbatlas_x509_authentication_database_user.go]: [nit] you don't need to pass an empty option here, `nil` is perfectly fine 
```suggestion
		certificates, _, err := conn.X509AuthDBUsers.GetUserCertificates(ctx, projectID, username, nil)
```

Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_x509_authentication_database_user.go]: I do have this pattern in resources/tests as well I have removed

**Review by gssbzn — APPROVED** [review]
LGTM thanks for those changes

**Pull request merged**